sklearn/preprocessing/imputation.py

```diff
@@ -299,6 +312,22 @@ def _dense_fit(self, X, strategy, missing_values, axis):
 
             return most_frequent
 
+        # KNN
+        elif strategy == "knn":
+
+            if axis == 1:
+                X = X.copy().transpose()
+
+            full_data = X[np.logical_not(mask.any(1))]
+            if full_data.size == 0:
+                raise ValueError("There is no row with complete data!")
+            if full_data.shape[0] < self.kneighbor:
+                raise ValueError("There are at most %d neighbors!" %(full_data.shape[0]))
```

Comment:
I think it should be ``"There are only %d complete data points, but n_neighbors=%d"``